uniflow/op/model/model_server.py

```diff
@@ -687,13 +673,59 @@ def _postprocess(self, data: List[str]) -> List[str]:
         """
         return data
 
-    def _get_provider(self) -> str:
-        return self._model_config.model_name.split(".")[0]
-
     def enforce_stop_tokens(self, text: str, stop: List[str]) -> str:
         """Cut off the text as soon as any stop words occur."""
         return re.split("|".join(stop), text, maxsplit=1)[0]
 
+    @abstractmethod
+    def prepare_input(
+        self, provider: str, prompt: str, model_kwargs: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """
+        Prepare the input for the model.
+        """
+        pass
+
+    @abstractmethod
+    def prepare_output(self, provider: str, response: Any) -> str:
+        """
+        Prepares the output based on the provider and response.
+        """
+        pass
+
+    @abstractmethod
+    def __call__(self, data: List[str]) -> List[str]:
+        """
+        Run model.
+        """
+        pass
```

Comment:
should we use raise exception to force people to implement these methods or `pass` is used as intended?

Reply:
I think raising NotImplementedError here is a better idea.